File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/Alert.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.annotations.ApiModel;
import lombok.*;

import java.time.LocalDate;

@ApiModel(description = "Alert")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString
@Data
public class Alert {
    private Long alertId;
    private Long bookingId;
    private String offenderNo;
    private String alertType;
    private String alertTypeDescription;
    private String alertCode;
    private String alertCodeDescription;
    private String comment;
    private LocalDate dateCreated;
    private LocalDate dateExpires;
    private boolean expired;
    private boolean active;
    private String addedByFirstName;
    private String addedByLastName;
    private String expiredByFirstName;
    private String expiredByLastName;
    private int ranking;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/Alert.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import java.io.Serializable
import java.time.LocalDate

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Alert(

  val alertCode: String,
  var dateCreated: LocalDate,
  val activeFrom: LocalDate? = null,
  var dateExpires: LocalDate? = null,
  val active: Boolean,
) : Serializable {

  fun isExpired(): Boolean {
    if (this.dateExpires != null) {
      return this.dateExpires!!.isBefore(LocalDate.now())
    }
    return false
  }

  companion object {
    private const val serialVersionUID: Long = 1
  }
}
